tests/unit/query_expression_evaluator.cpp

```diff
@@ -2386,4 +2454,38 @@ TYPED_TEST(FunctionTest, Duration) {
   EXPECT_EQ(this->EvaluateFunction("DURATION", "P3DT4H5M6.100110S").ValueDuration(),
             memgraph::utils::Duration({3, 4, 5, 6, 100, 110}));
 }
+
+TYPED_TEST(FunctionTest, ZonedDateTime) {
+  const auto date_parameters = memgraph::utils::DateParameters{2024, 6, 22};
+  const auto local_time_parameters = memgraph::utils::LocalTimeParameters{12, 6, 3, 0, 0};
+  const auto zdt = memgraph::utils::ZonedDateTime(
+      {date_parameters, local_time_parameters, memgraph::utils::Timezone("America/Los_Angeles")});
+  EXPECT_EQ(this->EvaluateFunction("DATETIME", "2024-06-22T12:06:03[America/Los_Angeles]").ValueZonedDateTime(), zdt);
+
+  const auto map_param = TypedValue(std::map<std::string, TypedValue>{{"year", TypedValue(2024)},
+                                                                      {"month", TypedValue(6)},
+                                                                      {"day", TypedValue(22)},
+                                                                      {"hour", TypedValue(12)},
+                                                                      {"minute", TypedValue(6)},
+                                                                      {"second", TypedValue(3)},
+                                                                      {"millisecond", TypedValue(0)},
+                                                                      {"microsecond", TypedValue(0)},
+                                                                      {"timezone", TypedValue("America/Los_Angeles")}});
+  EXPECT_EQ(this->EvaluateFunction("DATETIME", map_param).ValueZonedDateTime(), zdt);
+
+  const auto one_sec_in_microseconds = 1000000;
+  const auto today = memgraph::utils::CurrentZonedDateTime();
+  EXPECT_NEAR(this->EvaluateFunction("DATETIME").ValueZonedDateTime().SysMicrosecondsSinceEpoch().count(),
+              today.SysMicrosecondsSinceEpoch().count(), one_sec_in_microseconds);
+  EXPECT_EQ(this->EvaluateFunction("DATETIME", TypedValue(std::map<std::string, TypedValue>{})).ValueZonedDateTime(),
+            memgraph::utils::ZonedDateTime({{}, {}, memgraph::utils::DefaultTimezone()}));
+
+  EXPECT_THROW(this->EvaluateFunction("DATETIME", "{}"), memgraph::utils::BasicException);
+  EXPECT_THROW(
+      this->EvaluateFunction("DATETIME", TypedValue(std::map<std::string, TypedValue>{{"hours", TypedValue(1970)}})),
+      QueryRuntimeException);
+  EXPECT_THROW(
+      this->EvaluateFunction("DATETIME", TypedValue(std::map<std::string, TypedValue>{{"seconds", TypedValue(1970)}})),
```

Comment:
What will happen if you will pass just "year" or "month" or "day" or "millisecond" or "microsecond"?

Reply:
In that case, the created zoned datetime value will use the default values for all other properties. I’ll add a test to verify that.